server/ruleset.cpp

```diff
@@ -3527,11 +3527,6 @@ static bool load_ruleset_terrain(struct section_file *file,
         ok = FALSE;
         break;
       }
-
-      if (!ok) {
-        break;
-      }
-
```

Comment:
False positive, `ok` can be set to `false` two lines above.

Reply:
2 lines above is         
 ok = FALSE;
        break;
and it break iteration loop and never reaches that if